query.go

```diff
@@ -839,7 +845,13 @@ func (s *ChainService) GetBlock(blockHash chainhash.Hash,
 	// handleResp will be called for each message received from a peer. It
 	// will be used to signal to the work manager whether progress has been
 	// made or not.
-	handleResp := func(req, resp wire.Message, peer string) query.Progress {
+	handleResp := func(req, resp wire.Message, peer string, quit <-chan struct{}) query.Progress {
+		select {
+		case <-quit:
+			return noProgress
```

Comment:
same here, rather move this to the select on L #940 where things might actually hang